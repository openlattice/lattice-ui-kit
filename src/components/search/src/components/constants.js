import { List, Map, fromJS } from 'immutable';

const mockFilterFields = [
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'reportType',
    label: 'Report type',
    options: ['Crisis Template', 'Follow-up'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'badges',
    label: 'Badges',
    options: ['Officer Safety', 'Substance use'],
  },
  {
    filterCallback: (searchResult, filter) => {
      // if (filter.value === 'All') return true;
      // return searchResult.getIn([filter.id, 0]) === filter.value;
      return true;
    },
    id: 'submitter',
    label: 'Submitter',
    options: [
      'squidward@bubblebowl.com',
      'smitty@werbenjagermanjensen.com',
      'wumbology@thestudyofwumbo.com',
      'enigma@innermachinations.com'
    ],
  },
];

const mockSearchResultsForReports = List([
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety'],
    submitter: 'squidward@bubblebowl.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'squidward@bubblebowl.com',
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'smitty@werbenjagermanjensen.com'
  }),
  Map({
    reportType: 'Follow-up',
    badges: ['Substance use'],
    submitter: 'wumbology@thestudyofwumbo.com'
  }),
  Map({
    reportType: 'Crisis Template',
    badges: ['Officer Safety', 'Substance use'],
    submitter: 'enigma@innermachinations.com'
  }),
  Map({
    reportType: 'Follow-up',
    submitter: 'enigma@innermachinations.com'
  }),
]);

const mockSearchResultsForPeople = List([
  Map({
    lastName: 'Werbenjagermanjensen',
    firstName: 'Smitty',
    middleName: '',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Fish',
    dob: '02/22/2002',
    identifier: '365ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Thornton',
    firstName: 'Cameron',
    middleName: 'Orville',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Black / African American',
    dob: '01/05/1982',
    identifier: '565ba027-0c71-47d1-a5f4-61346ed96d8c',
  }),
  Map({
    lastName: 'Gerner',
    firstName: 'Timothy',
    middleName: 'Pernell',
    sex: 'M',
    gender: 'M',
    ethnicity: 'Native Hawaiian or Other Pacific Islander',
    dob: '10/02/1991',
    identifier: '465ba027-0c71-47d1-a5f4-61346ed96d8c',
  })
]);

export {
  mockFilterFields,
  mockSearchResultsForReports,
  mockSearchResultsForPeople
};
