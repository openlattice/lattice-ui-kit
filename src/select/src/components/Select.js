import React from 'react';
import ReactSelect, { createFilter } from 'react-select';
import { mergeDeep } from 'immutable';

import SelectController from './SelectController';
import { selectStyles } from '../../../style/select';

export const defaultProps = {
  filterOption: createFilter({ ignoreAccents: false }),
  menuPlacement: 'auto',
  styles: selectStyles,
};

/* eslint-disable react/jsx-props-no-spreading */
const Select = (props) => (
  <SelectController
      {...props} // eslint-disable-line indent
      render={((selectProps) => <ReactSelect {...mergeDeep(defaultProps, selectProps)} />)} />
);
/* eslint-enable */

export default Select;
