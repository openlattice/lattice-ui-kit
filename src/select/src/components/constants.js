export const LATTICE_SELECT = 'lattice-select';
export const OPTIONS = [
  { label: '0', value: 'zero' },
  { label: '1', value: 'one' },
  { label: '2', value: 'two' },
  { label: '3', value: 'three' },
  { label: '4', value: 'four' },
];
