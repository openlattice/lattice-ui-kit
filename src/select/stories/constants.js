// @flow

/* eslint-disable max-len */
const REACT_SELECT_USAGE :string = 'Select and Creatable components are simple wrappers that pass OpenLattice styling to their respective react-select components. All props provided to these wrappers are passed through allowing full use of react-select functionality.';

const stateOptions = [
  { value: 'AL', label: 'Alabama' },
  { value: 'AK', label: 'Alaska' },
  { value: 'AS', label: 'American Samoa' },
  { value: 'AZ', label: 'Arizona' },
  { value: 'AR', label: 'Arkansas' },
  { value: 'CA', label: 'California' },
  { value: 'CO', label: 'Colorado' },
  { value: 'CT', label: 'Connecticut' },
  { value: 'DE', label: 'Delaware' },
  { value: 'DC', label: 'District Of Columbia' },
  { value: 'FM', label: 'Federated States Of Micronesia' },
  { value: 'FL', label: 'Florida' },
  { value: 'GA', label: 'Georgia' },
  { value: 'GU', label: 'Guam' },
  { value: 'HI', label: 'Hawaii' },
  { value: 'ID', label: 'Idaho' },
  { value: 'IL', label: 'Illinois' },
  { value: 'IN', label: 'Indiana' },
  { value: 'IA', label: 'Iowa' },
  { value: 'KS', label: 'Kansas' },
  { value: 'KY', label: 'Kentucky' },
  { value: 'LA', label: 'Louisiana' },
  { value: 'ME', label: 'Maine' },
  { value: 'MH', label: 'Marshall Islands' },
  { value: 'MD', label: 'Maryland' },
  { value: 'MA', label: 'Massachusetts' },
  { value: 'MI', label: 'Michigan' },
  { value: 'MN', label: 'Minnesota' },
  { value: 'MS', label: 'Mississippi' },
  { value: 'MO', label: 'Missouri' },
  { value: 'MT', label: 'Montana' },
  { value: 'NE', label: 'Nebraska' },
  { value: 'NV', label: 'Nevada' },
  { value: 'NH', label: 'New Hampshire' },
  { value: 'NJ', label: 'New Jersey' },
  { value: 'NM', label: 'New Mexico' },
  { value: 'NY', label: 'New York' },
  { value: 'NC', label: 'North Carolina' },
  { value: 'ND', label: 'North Dakota' },
  { value: 'MP', label: 'Northern Mariana Islands' },
  { value: 'OH', label: 'Ohio' },
  { value: 'OK', label: 'Oklahoma' },
  { value: 'OR', label: 'Oregon' },
  { value: 'PW', label: 'Palau' },
  { value: 'PA', label: 'Pennsylvania' },
  { value: 'PR', label: 'Puerto Rico' },
  { value: 'RI', label: 'Rhode Island' },
  { value: 'SC', label: 'South Carolina' },
  { value: 'SD', label: 'South Dakota' },
  { value: 'TN', label: 'Tennessee' },
  { value: 'TX', label: 'Texas' },
  { value: 'UT', label: 'Utah' },
  { value: 'VT', label: 'Vermont' },
  { value: 'VI', label: 'Virgin Islands' },
  { value: 'VA', label: 'Virginia' },
  { value: 'WA', label: 'Washington' },
  { value: 'WV', label: 'West Virginia' },
  { value: 'WI', label: 'Wisconsin' },
  { value: 'WY', label: 'Wyoming' },
];

const paragraph = `
October arrived, spreading a damp chill over the grounds and into the castle. Madam Pomfrey, the nurse, was kept busy by a sudden spate of colds among the staff and students. Her Pepperup potion worked instantly, though it left the drinker smoking at the ears for several hours afterward. Ginny Weasley, who had been looking pale, was bullied into taking some by Percy. The steam pouring from under her vivid hair gave the impression that her whole head was on fire. Raindrops the size of bullets thundered on the castle windows for days on end; the lake rose, the flower beds turned into muddy streams, and Hagrid's pumpkins swelled to the size of garden sheds. Oliver Wood's enthusiasm for regular training sessions, however, was not dampened, which was why Harry was to be found, late one stormy Saturday afternoon a few days before Halloween, returning to Gryffindor Tower, drenched to the skin and splattered with mud.
`;

export {
  REACT_SELECT_USAGE,
  paragraph,
  stateOptions,
};
