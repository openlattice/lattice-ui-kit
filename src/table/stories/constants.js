const TABLE_HEADERS = [
  { key: 'name', label: 'Subject Name', cellStyle: { width: '50%' } },
  { key: 'dob', label: 'Date of Birth' },
  { key: 'manager', label: 'Manager' },
  { key: 'lastUpdated', label: 'Last Updated' },
  { key: 'id', label: 'ID', sortable: false },
];

const TABLE_DATA = [
  {
    name: 'Eugene Krabs',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-09-01',
    id: '1'
  },
  {
    name: 'Squidward Tentacles',
    dob: '',
    manager: 'Smitty',
    lastUpdated: '2019-08-29',
    id: '2'
  },
  {
    name: 'Sandy Cheeks',
    dob: '1987-11-17',
    manager: 'Patrick',
    lastUpdated: '2019-07-30',
    id: '3'
  },
  {
    name: 'Larry the Lobster',
    dob: '1975-06-23',
    manager: 'Patchy',
    lastUpdated: '2019-08-16',
    id: '4'
  },
  {
    name: 'Sheldon Plankton',
    dob: '1942-11-30',
    manager: 'Spongebob',
    lastUpdated: '2019-08-20',
    id: '5'
  },
  {
    name: 'Mrs. Puff',
    dob: '',
    manager: 'Spongebob',
    lastUpdated: '2019-08-21',
    id: '6'
  },
  {
    name: 'Flying Dutchman',
    dob: '1678-04-20',
    manager: 'Spongebob',
    lastUpdated: '2019-06-21',
    id: '7'
  },
];

export {
  TABLE_HEADERS,
  TABLE_DATA,
};
