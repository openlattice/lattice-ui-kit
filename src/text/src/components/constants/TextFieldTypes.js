/*
 * @flow
 */

type TextFieldTypeEnum = {|
  TEXT :'text';
  TEXT_AREA :'textarea';
|};
type TextFieldType = $Values<TextFieldTypeEnum>;

// TODO: look into using Immutable.Map() or other possible "enum" libraries
const TextFieldTypes :TextFieldTypeEnum = Object.freeze({
  TEXT: 'text',
  TEXT_AREA: 'textarea',
});

export default TextFieldTypes;
export type {
  TextFieldType,
  TextFieldTypeEnum,
};
