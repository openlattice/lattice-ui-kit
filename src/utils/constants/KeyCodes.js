/*
 * @flow
 */

type KeyCodeEnum = {|
  ENTER :'Enter';
  ESCAPE :'Escape';
|};
type KeyCode = $Values<KeyCodeEnum>;

// TODO: look into using Immutable.Map() or other possible "enum" libraries
const KeyCodes :KeyCodeEnum = Object.freeze({
  ENTER: 'Enter',
  ESCAPE: 'Escape',
});

export default KeyCodes;
export type {
  KeyCode,
  KeyCodeEnum,
};
