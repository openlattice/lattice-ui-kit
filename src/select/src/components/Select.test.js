import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Select from './Select';
import { OPTIONS } from './constants';

describe('Select', () => {

  it('render matches snapshot', () => {
    const tree = shallow(<Select />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('clicking should toggle menu', () => {
    const wrapper = mount(<Select />);
    expect(wrapper.find('Menu').exists()).toBeFalsy();
    wrapper.find('div.lattice-select__dropdown-indicator').simulate('mouseDown', { button: 0 });
    expect(wrapper.find('Menu').exists()).toBeTruthy();
  });

  it('single > should show controlled value', () => {
    const wrapper = mount(
      <Select
          options={OPTIONS}
          value={OPTIONS[0]} />
    );
    const actualValue = wrapper.find('Control').get(0).props.getValue();
    const expectedValue = [OPTIONS[0]];
    expect(actualValue).toEqual(expectedValue);
  });

  it('multi > should show controlled value', () => {
    const wrapper = mount(
      <Select
          isMulti
          options={OPTIONS}
          value={[OPTIONS[0], OPTIONS[1]]} />
    );
    const actualValue = wrapper.find('Control').get(0).props.getValue();
    const expectedValue = [
      OPTIONS[0],
      OPTIONS[1]
    ];
    expect(actualValue).toEqual(expectedValue);
  });

});
