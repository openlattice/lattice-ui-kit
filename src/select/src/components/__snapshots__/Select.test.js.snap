// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render matches snapshot 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  styles={
    Object {
      "clearIndicator": [Function],
      "container": [Function],
      "control": [Function],
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "indicatorsContainer": [Function],
      "menu": [Function],
      "menuPortal": [Function],
      "option": [Function],
      "singleValue": [Function],
    }
  }
/>
`;
